src/main/java/seedu/address/logic/commands/allocate/DeallocateCommand.java

```diff
@@ -152,9 +153,8 @@ public boolean equals(Object other) {
 
         // state check
         DeallocateCommand e = (DeallocateCommand) other;
-        assert employeeId != null;
         return eventIndex.equals(e.eventIndex)
-                && employeeId.equals(e.employeeId);
+                && (employeeId == e.employeeId || employeeId.equals(e.employeeId));
```

Comment:
why not just employeeId.equals(e.employeeId)

Reply:
cos have null~~